### PR TITLE
TS-9334 add support for switch nodes

### PR DESCRIPTION
**Pull request by anderswei**
Background: Ops are switching Cassandra nodes in our Cassandra cluster, and they found a huge amount of DB query errors. The driver is routing the traffic to a node that has been removed. 
This shouldn't happen because marina is actually very smart to detect those node changes, and should automatically skip those nodes. And we did tons of tests before the launch in local/dynamic env. 
After further digging, we find the root cause: in our test, what we are doing is: remove a node, add a node, it will automatically detect everything and everything is good. However, in Nikhil's operation, what happened in a node is removed and a new node is added at the same time. aka a switch, there is a bug in our code to handle those switches. 

Need to create a ticket in JIRA

### CODE REVIEW

File: src/marina_pool.erl [hunk 1]
```diff
@@ -208,11 +211,12 @@ stop_nodes(NodesToStop, Strategy, NewNodes) ->
     [
         begin
             NodeId = node_id(RpcAddress),
-            ets:delete(?MODULE, NodeId),
+            ets:insert(?MODULE, {NodeId, true}),
```
Comment (TC-aLi) [inline src/marina_pool.erl:269]: Why stop_nodes calls ets:insert and start_node calls ets:delete? And the data structures inserted and deleted don't match..

Comment (nitzanharel) [reply src/marina_pool.erl:269]: @anderswei ?

File: src/marina_pool_server.erl [hunk 2]
```diff
@@ -91,7 +91,7 @@ handle_msg(?MSG_PEER_WATCHER, #state {
             NodesToStart = Nodes -- OldNodes,
             NodesToStop = OldNodes -- Nodes,
             stop_nodes(NodesToStop, Strategy, Nodes),
-            start_nodes(Strategy, NodesToStart, OldNodes),
+            start_nodes(Strategy, NodesToStart, Nodes),
```
Comment (nitzanharel) [inline src/marina_pool_server.erl]: don't we need to start first and stop after?

**Comment by anderswei** [comment]
Tested together with Nikhil in env 5. It was working fine.

File: src/marina_pool.erl (hunk 1 above)
Comment (anderswei) [reply src/marina_pool.erl:269]: this ets table contains the *dead* node, not the running node.

**Review by TC-aLi — APPROVED** [review]

**Pull request merged**